python/faery/file_decoder.py

```diff
@@ -9,11 +9,7 @@
 from . import enums, events_stream, timestamp
 
 if typing.TYPE_CHECKING:
```

Comment:
Ah, this is interesting. I haven't noticed this before, but I guess this allows the user to omit types (and suffer the potential consequences)?

Reply:
This is actually to make static type checkers happy. At runtime, `typing.TYPE_CHECKING` is false and Python imports the functions' implementation from the compiled extension. However, this extension is a black box for static type checkers which need to use *.pyi* files instead. When statically checking the code, type checkers assume that `typing.TYPE_CHECKING` is true and they load types from the *pyi* interfaces.

Reply:
That explains it! Thank you